runtime/context.go

```diff
@@ -271,3 +296,97 @@ func (c *contextLogger) Warn(ctx context.Context, msg string, userFields ...zap.
 func (c *contextLogger) Check(lvl zapcore.Level, msg string) *zapcore.CheckedEntry {
 	return c.log.Check(lvl, msg)
 }
+
+// ContextMetrics emits metrics.
+type ContextMetrics interface {
+	GetOrAddInboundHTTPMetrics(ctx context.Context) *InboundHTTPMetrics
```

Comment:
To be consistent with https://github.com/uber/zanzibar/pull/499, do we still need the word `Inbound` in the name?

Reply:
Will change naming in #499,